providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/common/ConnectionState.java

```diff
@@ -0,0 +1,58 @@
+package dev.openfeature.contrib.providers.flagd.resolver.common;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * Represents the state of a connection, indicating whether it is connected,
+ * disconnected, or stale.
+ * <p>
+ * This class is immutable and uses the {@link lombok.AllArgsConstructor} annotation
+ * to generate a constructor with parameters for all fields.
+ * It also uses {@link lombok.Getter} to provide getter methods for the fields.
+ */
+@AllArgsConstructor
+public class ConnectionState {
```

Comment:
Currently, this class can handle two dedicated state properties. However, they could also be represented via enums, which may be easier to read/maintain. We currently have four states (Disconnected, Connected, Stale, and Error). Would it be worth changing this to enums? We could skip handling different state variables and just utilize the enums.